### PR TITLE
[Docs] Update install guide and README

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

We have changed our development workflow and now we develop on the `dev` branch. Therefore we need to update our install guide.

## Modification

1. Update install guide.
2. Update README, use readthedocs link preferentially. And add new repos.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,3 +141,5 @@ We wish that the toolbox and benchmark could serve the growing research communit
 - [MMFlow](https://github.com/open-mmlab/mmflow) OpenMMLab optical flow toolbox and benchmark.
 - [MMFewShot](https://github.com/open-mmlab/mmfewshot): OpenMMLab FewShot Learning Toolbox and Benchmark.
 - [MMHuman3D](https://github.com/open-mmlab/mmhuman3d): OpenMMLab 3D Human Parametric Model Toolbox and Benchmark.
+- [MMSelfSup](https://github.com/open-mmlab/mmselfsup): OpenMMLab self-supervised learning toolbox and benchmark.
+- [MMDeploy](https://github.com/open-mmlab/mmdeploy): OpenMMLab Model Deployment Framework.
```
Comment (Ezra-Yu) [inline README.md:145]: ADD:
```
- [MMRazor](https://github.com/open-mmlab/mmrazor): OpenMMLab Model Compression Toolbox and Benchmark.
```

File: README.md [hunk 2]
```diff
@@ -85,18 +85,18 @@ Results and models are available in the [model zoo](docs/en/model_zoo.md).
...
-- [customizing runtime settings](docs/en/tutorials/runtime.md)
+- [Learn about Configs](https://mmclassification.readthedocs.io/en/latest/tutorials/config.html)
+- [Fine-tune Models](https://mmclassification.readthedocs.io/en/latest/tutorials/finetune.html)
+- [Adding New Dataset](https://mmclassification.readthedocs.io/en/latest/tutorials/new_dataset.html)
+- [Customizie Data Pipeline](https://mmclassification.readthedocs.io/en/latest/tutorials/data_pipeline.html)
+- [Add New Modules](https://mmclassification.readthedocs.io/en/latest/tutorials/new_modules.html)
+- [Customizie Schedule](https://mmclassification.readthedocs.io/en/latest/tutorials/schedule.html)
+- [Customizie Runtime Settings](https://mmclassification.readthedocs.io/en/latest/tutorials/runtime.html)
 
 Colab tutorials are also provided. To learn about MMClassification Python API, you may preview the notebook [here](https://github.com/open-mmlab/mmclassification/blob/master/docs/en/tutorials/MMClassification_python.ipynb) or directly [run](https://colab.research.google.com/github/open-mmlab/mmclassification/blob/master/docs/en/tutorials/MMClassification_python.ipynb) on Colab.
```
Comment (Ezra-Yu) [inline README.md:101]: Colab tutorials are also provided:

Learn about **MMClassification Python API**: [notebook](xxxx) or  directly [run](xxxx);
Learn about **MMClassification  shell tool**: [notebook](xxxx) or directly [run](xxxx);

may be simpler.

File: README.md [hunk 3]
```diff
@@ -85,18 +85,18 @@ Results and models are available in the [model zoo](docs/en/model_zoo.md).
...
-- [learn about configs](docs/en/tutorials/config.md)
-- [finetuning models](docs/en/tutorials/finetune.md)
-- [adding new dataset](docs/en/tutorials/new_dataset.md)
-- [designing data pipeline](docs/en/tutorials/data_pipeline.md)
-- [adding new modules](docs/en/tutorials/new_modules.md)
-- [customizing schedule](docs/en/tutorials/schedule.md)
-- [customizing runtime settings](docs/en/tutorials/runtime.md)
+- [Learn about Configs](https://mmclassification.readthedocs.io/en/latest/tutorials/config.html)
+- [Fine-tune Models](https://mmclassification.readthedocs.io/en/latest/tutorials/finetune.html)
+- [Adding New Dataset](https://mmclassification.readthedocs.io/en/latest/tutorials/new_dataset.html)
```
Comment (Ezra-Yu) [inline README.md:95]: Learn About Configs
Add New Dataset

Keep the format consistent

**Review by Ezra-Yu — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (mzr1996) [reply README.md]: Use lower case for `about` is in title case format.

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Pull request merged**